src/testdir/test_statusline.vim

```diff
@@ -251,6 +251,18 @@ func Test_statusline()
   call assert_match('^vimLineComment\s*$', s:get_statusline())
   syntax off
 
+  "%{: evaluates enxpressions present in result of %{} expression
+  func! Inner_eval()
+    return '%n some other text'
```

Comment:
What happens if this line is the next one?
```vim
    return '%{Outer_eval()}'
```

Reply:
Now nothing special will happen . When you asked it would've gotten stuck because of infinite recursion . But now evaluation stops when 100 level depth is reached.

Reply:
Oh, thanks. That's exactly what I was afraid of.